presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlDistributedQueries.java

```diff
@@ -168,6 +169,42 @@ public void testMySqlCreatedParameterizedVarchar()
         assertUpdate("DROP TABLE mysql_test_parameterized_varchar");
     }
 
+    @Test
+    public void testMySqlTinyInt1Describe()
+            throws Exception
+    {
+        execute("CREATE TABLE tpch.mysql_test_tinyint1 (" +
+                "c_tinyint tinyint(1)" +
+                ")");
+
+        MaterializedResult actual = computeActual("SHOW COLUMNS FROM mysql_test_tinyint1");
+
+        MaterializedResult expected = resultBuilder(getSession(), TINYINT)
+                .row("c_tinyint", "tinyint", "").build();
+
+        assertEquals(actual, expected);
+
+        assertUpdate("DROP TABLE mysql_test_tinyint1");
+    }
+
+    @Test
+    public void testMySqlTinyInt1Select()
+            throws Exception
+    {
+        execute("CREATE TABLE tpch.mysql_test_tinyint1 (" +
+                "c_tinyint tinyint(1)" +
+                ")");
+        execute("INSERT INTO tpch.mysql_test_tinyint1 VALUES(127),(-128)");
+
+        MaterializedResult materializedRows = computeActual("SELECT * from mysql_test_tinyint1 WHERE c_tinyint = 127");
+        assertEquals(materializedRows.getRowCount(), 1);
+        MaterializedRow row = getOnlyElement(materializedRows);
```

Comment:
`getOnlyColumnAsSet()` assumes there're no duplicates in rows (set). I don't think it's what we need here. Currently it does not matter but what for introducing such a constraint?
What's more, Java won't like casting Byte to String (`getOnlyColumnAsSet()` casts results to String):
```java
java.lang.ClassCastException: java.lang.Byte cannot be cast to java.lang.String
```

I'd leave it as is.